megamek/src/megamek/client/ui/swing/ClientGUI.java

```diff
@@ -1850,12 +1851,11 @@ public void loadPreviewImage(JLabel bp, Entity entity) {
     }
 
     public void loadPreviewImage(JLabel bp, Entity entity, IPlayer player) {
-        Image camo;
-        if (entity.getCamoFileName() != null) {
-            camo = bv.getTilesetManager().getEntityCamo(entity);
-        } else {
-            camo = bv.getTilesetManager().getPlayerCamo(player);
+        Image camo = CamoManager.getPlayerCamoImage(player);
+        if (entity.getCamoCategory() != null && !entity.getCamoCategory().equals(IPlayer.NO_CAMO)) {
+            camo = CamoManager.getEntityCamoImage(entity);
         }
+        // This seems unnecessary as the CamoManager will return an image for a playercolor:
         int tint = PlayerColors.getColorRGB(player.getColorIndex());
```

Comment:
?

Reply:
The following code can probably be simplified. But this PR is already much bigger than I originally planned and I needed to stop at some place. This is the place.